tests/e2e/tests/mixer/mixer_test.go

```diff
@@ -0,0 +1,568 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package mixer defines integration tests that validate working mixer
+// functionality in context of a test Istio-enabled cluster.
+package mixer
+
+import (
+	"bytes"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/golang/glog"
+	dto "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+
+	"istio.io/istio/tests/e2e/framework"
+	"istio.io/istio/tests/e2e/util"
+)
+
+const (
+	bookinfoYaml             = "demos/apps/bookinfo/bookinfo.yaml"
+	rulesDir                 = "demos/apps/bookinfo"
+	rateLimitRule            = "mixer-rule-ratings-ratelimit.yaml"
+	denialRule               = "mixer-rule-ratings-denial.yaml"
+	newTelemetryRule         = "mixer-rule-additional-telemetry.yaml"
+	routeAllRule             = "route-rule-all-v1.yaml"
+	routeReviewsVersionsRule = "route-rule-reviews-v2-v3.yaml"
+	emptyRule                = "mixer-rule-empty-rule.yaml"
+
+	mixerAPIPort        = 9094
+	mixerPrometheusPort = 42422
+
+	targetLabel       = "target"
+	responseCodeLabel = "response_code"
+
+	global = "global"
+)
+
+type testConfig struct {
+	*framework.CommonConfig
+	gateway  string
+	rulesDir string
+}
+
+var (
+	tc    *testConfig
+	rules = []string{rateLimitRule, denialRule, newTelemetryRule, routeAllRule, routeReviewsVersionsRule, emptyRule}
+)
+
+func (t *testConfig) Setup() error {
+	t.gateway = "http://" + tc.Kube.Ingress
+	for _, rule := range rules {
```

Comment:
true. this code copies to t.rulesDir, which makes it easier to reference tc.rulesDir later in the create rules blocks. i'll update the code to make that clearer.